### ISSUE TITLE
Condition for visibility of the "Propose a Governance Action" button in the govtool header

### ISSUE THREAD

**Issue by teske00** [IntersectMBO/govtool#1199]
## Context & versions

After the implementation of wallet singData for the needs of user login to PDF (https://github.com/IntersectMBO/govtool-proposal-discussion/issues/66), the condition for the visibility of the button in the header is not whether the wallet is connected, but whether the user is logged in to PDF. As the user's state is at the PDF level, the govtool wrapper cannot know the user's state on the PDF. In order for the button to remain in the header, it is necessary to move the PDF user state to the govtool wrapper level or move the button to PDF. By moving the state of the PDF user to the govtool wrapper, we make the wrapper dependent on the package, and instead of the package depending only on the state of the wallet, it also depends on other functions necessary for its functioning, which should be kept at the package level.

**Comment by teske00** [comment]
cc @kickloop @Ryun1

**Comment by l-br1** [comment]
could we move the button to the page (where it was actually supposed to be) so that it shows on the proposal discussion forum page without toughing the header?

And then do the same thing for the GA page (which needs the same button, OR a button with the same functionality AKA opening the proposal submission flow).

CC: @kickloop @m-i-k-e @MSzalowski

**Comment by teske00** [comment]
@l-br1 I don't know if it's the same button, our button takes us to the flow to create a PDF proposal. That button had no logic, just to navigate to the creation flow. But that flow can only be accessed if the user is logged in to PDF. I'd say we're not talking about the same button
![image](https://github.com/IntersectMBO/govtool/assets/63746943/78c041de-deca-4542-846d-a950b14f0f73)

**Comment by vpaunovic** [comment]
Yeah, I agree with @teske00 .. That button is for opening the flow for creation of proposal. And for sumission part, there is other button, inside proposal details.

**Comment by MSzalowski** [comment]
From the implementation side - +1 for moving that button directly to the PDF Pillar as it doesn't need any `govtool-wrapper` related logic. 
From the UX perspective I am not the one to decide I guess. If that button needs the condition that is kept under PDF pillar, we can consume that button delivered directly from PDF pillar package that includes such logic and still display it in a wrapper, not recommended due to mixing up the logic part, but achievable.

What is the decision?
@l-br1 @teske00

**Comment by l-br1** [comment]
@teske00 yes, that is the same button. The 'propose Governance action' button needs to be both on the proposal page and on the GA list. This is because on the GA list users need to have the call to action to start the process to create a GA.

So same button, same cta, same flow. Now the position of the button can be different and the button can be a different button as long as the outcome is the same

**Comment by teske00** [comment]
@l-br1 With this, we return to the same state as if the button was in the header, where it is necessary to transfer the user's state to the wrapper, which we try to avoid because this is how we make the wrapper dependent on the PDF package. If we leave the user state to be only on PDF, then that button can only lead to the PDF page, but not to the flow for creating the proposal. Or we create a mutual dependency between the package and the wrapper by putting the user state in the wrapper, which removes the feature of the package being independent except for the wallet state, which was the idea from the beginning

**Comment by l-br1** [comment]
question, what happens if a user gets into govtool via a direct link to a proposal or the proposal creation process? What happens if the user is already in (so wallet connected) but in another page, what happens if the user is no connected with the wallet?

What I am trying to understand is, if I get to the URL of the proposal creation process (which also is the ga creation process) so I need to pass from the proposal discussion page for it to work? If not, can the button on the GA list work in the same way.

p.s. if this requires more thinking, the button on GA list is needed but not a blocker for launch cc: @ValMBO @bosko-m

**Comment by teske00** [comment]
@l-br1 The problem is that after the implementation of wallet signData, in order to ensure the authenticity of the wallet before logging in to PDF, the status of the connected wallet is not enough, but whether the login was successful, that is, whether we received the JWT for that user from the backend. If you use the link to go directly to the page for creating a proposal, it is currently set to redirect you to the PDF home page where there is a list of proposals, because the creation process is only possible if we have a logged-in user in the state in PDF

**Comment by bosko-m** [comment]
This should be a wider discussion that needs to happen on how we are going to handle and display username within the whole GovTool app.

**Comment by m-i-k-e** [comment]
Username seems to have been added to PDF in a way that assumes it's a separate application and not part of a larger GovTool.

**Comment by bosko-m** [comment]
Correct, hence the reason for a wider discussion.

**Comment by kneerose** [comment]
- The new design for 'Verify Your Identity' is described [here](https://github.com/intersectMBO/govtool/issues/3549).
- The idea to display the username is mentioned in this [ticket](https://github.com/intersectMBO/govtool/issues/3221).

- Propose a Governance Action" button on the Governance Action page:

  ![Image](https://github.com/user-attachments/assets/7c9ebb3c-1a21-4140-96a3-349949799338)

- Propose a Governance Action" button on the Proposal Discussion page:

  ![Image](https://github.com/user-attachments/assets/7dbf50cc-6051-4742-9630-ccf2ad8774c7)

If this information covered everything discussed above, then this ticket is ready to close.
CC: @bosko-m
